Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Sos.hs

```diff
@@ -50,22 +103,88 @@ newtype SosRes = SosRes
+    OneWaySpecialZoneDetails bDetails -> pure $ mkAddressEntity bDetails.toLocation.address
+  let rideInfo =
+        Ticket.RideInfo
+          { rideShortId = ride.shortId.getShortId,
+            customerName = Just $ (fromMaybe "" person.firstName) <> " " <> (fromMaybe "" person.lastName),
+            customerPhoneNo = Just $ countryCode <> mobileNumber,
+            driverName = Just ride.driverName,
+            driverPhoneNo = Just ride.driverMobileNumber,
+            vehicleNo = ride.vehicleNumber,
+            status = show booking.status,
+            rideCreatedAt = ride.createdAt,
+            pickupLocation = mkAddressEntity booking.fromLocation.address,
+            dropLocation = Just toLocation,
+            fare = Nothing
+          }
+  _ <- createTicket person.merchantId (mkTicket person (Just phoneNumber) ride.trackingUrl rideInfo)
+  when (person.shareEmergencyContacts) $ do
+    emergencyContacts <- getDefaultEmergencyNumbers (personId, merchantId)
+    let sender = smsCfg.sender
+        trackLink = "nammayatri.in/t/" <> show ride.shortId
+    message <-
+      MessageBuilder.buildSOSAlertMessage person.merchantId $
+        MessageBuilder.BuildSOSAlertMessageReq
+          { userName = (fromMaybe "" person.firstName) <> " " <> (fromMaybe "" person.lastName),
+            rideLink = trackLink
+          }
+    for_ emergencyContacts.defaultEmergencyNumbers $ \emergencyContact -> do
+      let emergencyPhoneNumber = emergencyContact.mobileCountryCode <> emergencyContact.mobileNumber
+      Sms.sendSMS person.merchantId (Sms.SendSMSReq message emergencyPhoneNumber sender)
+        >>= Sms.checkSmsResult
+
   return $
     SosRes
       { sosId = sosDetails.id
       }
+  where
+    mkAddressEntity addr =
+      let addField field label = case field of
+            Just value -> T.concat [label, ": ", value, ", "]
+            Nothing -> ""
+       in T.concat
+            [ addField addr.street "Street",
+              addField addr.city "City",
+              addField addr.state "State",
+              addField addr.country "Country",
+              addField addr.building "Building",
+              addField addr.areaCode "Area Code",
+              addField addr.area "Area"
+            ]
 
-updateSosDetails :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r) => Id DSos.Sos -> Id Person.Person -> SosFeedbackReq -> m APISuccess.APISuccess
-updateSosDetails sosId personId req = do
+updateSosDetails ::
+  Id DSos.Sos -> (Id Person.Person, Id Merchant.Merchant) -> SosUpdateReq -> Flow APISuccess.APISuccess
+updateSosDetails sosId (personId, merchantId) req = do
   sosDetails <- runInReplica $ QSos.findById sosId >>= fromMaybeM (SosIdDoesNotExist sosId.getId)
-
   unless (personId == sosDetails.personId) $ throwError $ InvalidRequest "PersonId not same"
-
   void $ QSos.updateStatus sosId (req.status)
+  _ <- Ticket.updateTicket merchantId (Ticket.UpdateTicketReq req.comment sosId.getId "" (show req.status) "" "" "")
```

Comment:
Can you use records when make `Ticket.UpdateTicketReq`, because fields are not clear?
Also, `UpdateTicketReq` is interface type, so we can remove unused fields from this type (all these empty strings)